sage.py

```diff
@@ -481,1274 +423,920 @@ def getepisodes(action_seq, mcat, plot, debug=False):
+    print('Usage: sage.py {path/to/json/files} {experiment_name} {alert_filtering_window (def=1.0)} {alert_aggr_window (def=150)} {(start_hour,end_hour)[Optional]}')
     sys.exit()
 
-folder = sys.argv[1]
-expname = sys.argv[2]
-t = float(sys.argv[3])
-w = int(sys.argv[4])
-_s_,_e_ = 0, 100
+path_to_json_files = sys.argv[1]
+experiment_name = sys.argv[2]
+alert_filtering_window = float(sys.argv[3])
+alert_aggr_window = int(sys.argv[4])
+start_hour, end_hour = 0, 100
 if len(sys.argv) > 5:
-    #range_ = str(sys.argv[5])
     try:
-        #range_ = range_.replace('(', '').replace(')', '').split(',')
-        _s_ = float(sys.argv[5])#int(range_[0])
-        _e_ = float(sys.argv[6])#int(range_[1])
-        print('Filtering alerts. Only parsing from %d-th to %d-th hour (relative to the start of the alert capture)'%(_s_,_e_))
-    except:
+        start_hour = float(sys.argv[5])
+        end_hour = float(sys.argv[6])
+        print('Filtering alerts. Only parsing from %d-th to %d-th hour (relative to the start of the alert capture)' % (start_hour, end_hour))
+    except (ValueError, TypeError):
         print('Error parsing hour filter range')
         sys.exit()
 
-startTimes = [] # We cheat a bit: In case user filters to only see alerts from (s,e) range, 
-#                    we record the first alert just to get the real time-elapsed since first alert
-    
-outaddress = ""
+# We cheat a bit: In case user filters to only see alerts from (s,e) range,
+#   we record the first alert just to get the real time-elapsed since first alert
+start_times = []
+
 path_to_ini = "FlexFringe/ini/spdfa-config.ini"
 
-modelname = expname+'.txt'#'test-trace-uni-serGroup.txt'
-datafile = expname+'.txt'#'trace-uni-serGroup.txt'
-   
-path_to_traces = datafile
-
-print('------ Downloading the IANA port-service mapping ---------')
-port_services = load_IANA_mapping()
-
-print('----- Reading alerts ----------')
-(unparse, team_labels) = load_data(folder, t) # t = minimal window for alert filtering
-plt = plot_histogram(unparse, team_labels)
-plt.savefig('data_histogram-'+expname+'.png')
-print('------ Converting to episodes ---------')
-team_episodes,_ = aggregate_into_episodes(unparse, team_labels, step=w) # step = w
-print('\n---- Converting to episode sequences -----------')
-host_data  =  host_episode_sequences(team_episodes)
-print('----- breaking into sub-sequences and making traces----------')
-(alerts, keys) = break_into_subbehaviors(host_data)
-generate_traces(alerts, keys, datafile)
-
-
-print('------ Learning SPDFA ---------')
-# Learn S-PDFA
+path_to_traces = experiment_name + '.txt'
+ag_directory = experiment_name + 'AGs'
+
+print('------ Downloading the IANA port-service mapping ------')
+port_services = load_iana_mapping()
+
+print('------ Reading alerts ------')
+(team_alerts, team_labels) = load_data(path_to_json_files, alert_filtering_window)
+# plot_histogram(team_alerts, team_labels)
```

Comment:
I have commented this out, as it is not really needed for AGs and can always be uncommented if needed

Reply:
No, keep the plot_histogram() in. The plot should always be generated and serves as a quick check for whether the alerts were correctly parsed.

Reply:
Alright. Just that I remember that during RP (and also now) I was deleting these plots after every run and almost never looked at them. These plot was used in the main paper, so I thought it is only needed for this purpose